chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -634,6 +634,29 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
         Ok(())
     }
 
+    fn check_no_signature_size(&self, block: &Block) -> Result<(), CheckBlockTransactionsError> {
+        for tx in block.transactions() {
+            for signature in tx.signatures() {
+                match signature {
+                    common::chain::signature::inputsig::InputWitness::NoSignature(data) => {
+                        if let Some(inner_data) = data {
+                            if inner_data.len() > self.chain_config.max_no_signature_data_size() {
+                                return Err(
+                                    CheckBlockTransactionsError::NoSignatureDataSizeTooLarge(
+                                        inner_data.len(),
+                                        self.chain_config.max_no_signature_data_size(),
+                                    ),
+                                );
+                            }
```

Comment:
Tiny nitpick: This could be `ensure!`... saves one level of indentation

Reply:
another option is txs.iter().flat_map(signatures).find_map(match...)

Reply:
Went with ensure. Thanks.